R/utils.R

```diff
@@ -1923,35 +1923,43 @@ S4ToList.list <- function(object) {
   return(object)
 }
 
+#' Simplify segmentations by reducing the number of vertices
+#'
+#' @param coords A `Segmentation` object
+#' @param tol Numerical tolerance value to be used by the Douglas-Peuker algorithm
+#' @param topologyPreserve Logical determining if the algorithm should attempt to preserve the topology of the original geometry
+#'
+#' @return A `Segmentation` object with simplified segmentation vertices
+#' 
 #' @rdname Simplify
 #' @method Simplify Spatial
 #' @export
 #'
 Simplify.Spatial <- function(coords, tol, topologyPreserve = TRUE) {
-  if (!PackageCheck('rgeos', error = FALSE)) {
-    stop("'Simplify' requires rgeos to be installed", call. = FALSE)
+  if (!PackageCheck("sf", error = FALSE)) {
+    stop("'Simplify' requires sf to be installed", call. = FALSE)
   }
   class.orig <- class(x = coords)
+  coords.orig <- coords
   dest <- ifelse(
-    test = grepl(pattern = '^Spatial', x = class.orig),
+    test = grepl(pattern = "^Spatial", x = class.orig), 
     yes = class.orig,
-    no = grep(
-      pattern = '^Spatial',
-      x = .Contains(object = coords),
-      value = TRUE
-    )[1L]
-  )
-  coords <- rgeos::gSimplify(
-    spgeom = as(object = coords, Class = dest),
-    tol = as.numeric(x = tol),
-    topologyPreserve = isTRUE(x = topologyPreserve)
-  )
-  coords <- tryCatch(
-    expr = as(object = coords, Class = class.orig),
-    error = function(...) {
-      return(coords)
-    }
-  )
+    no = grep(pattern = "^Spatial", x = .Contains(object = coords), value = TRUE)[1L])
```

Comment:
Slight stylistic fix
```suggestion
    no = grep(pattern = "^Spatial", x = .Contains(object = coords), value = TRUE)[1L]
  )
```